File: mobile/src/app/models/perfumesearchrequest.dto.ts
import { GenderEnum } from './gender.enum'

export interface PerfumeSearchRequestDto {
  query?: string | null;
  brandId?: number | null;
  countryCode?: string | null;
  minRating?: number | null;
  gender?: GenderEnum | null;
  groupIds?: number[];
  page: number;
  pageSize: number;
}

File: backend/Core/DTOs/PerfumeSearchRequestDto.cs
using Core.Enums;

namespace Core.DTOs
{
    public sealed class PerfumeSearchRequestDto
    {
        public string? Query { get; init; }
        public int? BrandId { get; init; }
        public string? CountryCode { get; init; }
        public double? MinRating { get; init; }
        public GenderEnum? Gender { get; init; }
        public IReadOnlyList<int>? GroupIds { get; init; }
        public int Page { get; init; } = 1;
        public int PageSize { get; init; } = 25;
    }
}
